src/sempy_labs/tom/_model.py

```
@@ -3545,24 +3659,37 @@ def add_time_intelligence(
         for t in time_intel:
             t = t.capitalize()
             if t not in [time_intel_options]:
-                raise ValueError(f"{icons.red_dot} The '{t}' time intelligence variation is not supported. Valid options: {time_intel_options}.")
+                raise ValueError(
+                    f"{icons.red_dot} The '{t}' time intelligence variation is not supported. Valid options: {time_intel_options}."
+                )
 
         # Validate measure and extract table name
-        for m in self.all_measures():
-            if m.Name == measure_name:
-                table_name = m.Parent.Name
-
-        if table_name is None:
-            raise ValueError(f"{icons.red_dot} The '{measure_name}' is not a valid measure in the '{self._dataset}' semantic model within the '{self._workspace}' workspace.")
+        try:
+            table_name = next(
+                m.Parent.Name for m in self.all_measures() if m.Name == measure_name
+            )
+        except:
```

is this exception hit if the measure is not found?

if so, I'd rather have a list created, check if the len is zero and then throw. right now you're masking all other exceptions.